lms/djangoapps/mobile_api/users/tests.py

```diff
@@ -179,16 +191,22 @@ def test_courseware_access(self):
                 self.assertFalse(result['has_access'])
 
     @ddt.data(
-        ('next_week', ADVERTISED_START, ADVERTISED_START, "string"),
-        ('next_week', None, defaultfilters.date(NEXT_WEEK, "DATE_FORMAT"), "timestamp"),
-        ('next_week', '', defaultfilters.date(NEXT_WEEK, "DATE_FORMAT"), "timestamp"),
-        ('default_start_date', ADVERTISED_START, ADVERTISED_START, "string"),
-        ('default_start_date', '', None, "empty"),
-        ('default_start_date', None, None, "empty"),
+        ('next_week', ADVERTISED_START, ADVERTISED_START, "string", API_V05),
+        ('next_week', ADVERTISED_START, ADVERTISED_START, "string", API_V1),
+        ('next_week', None, defaultfilters.date(NEXT_WEEK, "DATE_FORMAT"), "timestamp", API_V05),
+        ('next_week', None, defaultfilters.date(NEXT_WEEK, "DATE_FORMAT"), "timestamp", API_V1),
+        ('next_week', '', defaultfilters.date(NEXT_WEEK, "DATE_FORMAT"), "timestamp", API_V05),
+        ('next_week', '', defaultfilters.date(NEXT_WEEK, "DATE_FORMAT"), "timestamp", API_V1),
+        ('default_start_date', ADVERTISED_START, ADVERTISED_START, "string", API_V05),
+        ('default_start_date', ADVERTISED_START, ADVERTISED_START, "string", API_V1),
+        ('default_start_date', '', None, "empty", API_V05),
+        ('default_start_date', '', None, "empty", API_V1),
+        ('default_start_date', None, None, "empty", API_V05),
+        ('default_start_date', None, None, "empty", API_V1),
```

Comment:
Would this be clearer using something like `itertools.product` so that you don't have to explicitly list the API versions for each type of date check?

Reply:
@cpennington 
I tried using itertools.product like the following:
```
itertools.product(
        ['next_week', 'default_start_date'],
        [ADVERTISED_START, None, ''],
        [ADVERTISED_START, defaultfilters.date(NEXT_WEEK, "DATE_FORMAT"), None],
        ['string', 'timestamp', 'empty'],
        [API_V05, API_V1],
)
```
But this resulted in many more permutations then the test was written to handle and caused a bunch of failures. Is there another way to keep the exact set of ddt.data but somehow expand the api versions so I don't have duplicate?